### DISCUSSION TITLE
How can I avoid printing output results when calling the predict method?

### DISCUSSION THREAD

**Discussion by montmejat** [ultralytics/ultralytics#1703] [General]
Every time I'm doing a detection using `model.predict(...)` I get something like this in the terminal:
```
0: 512x512 (no detections), 29.4ms
Speed: 2.3ms preprocess, 29.4ms inference, 4.6ms postprocess per image at shape (1, 3, 512, 512)
```

Any ways of avoiding printing this? 😄

Thanks!

**Comment by dh031200** [comment]
add `verbose=False`

like this:
`model.predict(source=frame, verbose=False)`

**Reply by montmejat** [reply]
Thanks! Works great

**Reply by glenn-jocher** [reply]
@dh031200 you can also set the environment variable `YOLO_VERBOSE=False`, which will silence ALL YOLO outputs.

**Comment by payanrayan** [comment]
import logging, sys

logger = logging.getLogger()
logger.setLevel(logging.INFO)
formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
handler = logging.StreamHandler(sys.stdout)
handler.setFormatter(formatter)
logger.addHandler(handler)
logging.getLogger("ultralytics").setLevel(logging.WARNING)

**Reply by payanrayan** [reply]
Put this code at the beginning of your code

**Reply by pderrenger** [reply]
Certainly! If you'd like to suppress the output from the `predict` method, you can adjust the logging level for the Ultralytics library. Here's a small snippet that sets the logging level to `WARNING`, which should prevent the usual output from `predict` while still showing any warnings or errors:

```python
import logging

# Set logging level to WARNING to suppress INFO logs
logging.getLogger('ultralytics').setLevel(logging.WARNING)

# Your code here
# model.predict(...)
```

By adding this at the beginning of your script, you should no longer see the informational output during prediction. 😊 Let me know if you need any further assistance!
